diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/CustomLabel.kt

```diff
@@ -0,0 +1,42 @@
+package org.cqfn.diktat.ruleset.rules
+
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.EmitType
+import org.cqfn.diktat.ruleset.constants.Warnings.CUSTOM_LABEL
+
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType.CALL_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.FOR
+import com.pinterest.ktlint.core.ast.ElementType.LABEL_QUALIFIER
+import com.pinterest.ktlint.core.ast.ElementType.REFERENCE_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.WHILE
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.psi.psiUtil.parents
+
+/**
+ * Rule that checks using custom label
+ */
+class CustomLabel(private val configRules: List<RulesConfig>) : Rule("custom-label") {
+    private var isFixMode: Boolean = false
+    private val forEachReference = listOf("forEach", "forEachIndexed")
+    private val labels = listOf("@loop", "@forEach", "@forEachIndexed")
+    private val nestedParent = listOf(FOR, WHILE)
+    private lateinit var emitWarn: EmitType
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       emit: EmitType) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        if (node.elementType == LABEL_QUALIFIER && node.text !in labels) {
+            val nestedCount = node.parents().count {
+                it.elementType in nestedParent ||
+                        (it.elementType == CALL_EXPRESSION && it.findChildByType(REFERENCE_EXPRESSION)?.text in forEachReference)
```

Comment:
I don't think this is necessary: for me using `forEach` inside raw loops seems like a bad idea anyway. And I don't see a test for this case.